client/src/webserver.rs

```diff
@@ -151,6 +153,15 @@ async fn callback<'a>(
 
                     let mut config = state.config.clone();
                     config.token = Some(token.clone());
+
+                    // Ingest should default to 10 days expiry, so if we don't have it set that
+                    config.token_expires_at = match params.get("expires_at") {
+                        None => Some(Utc::now().date_naive().add(Days::new(10))),
+                        Some(expires) => Some(
+                            chrono::NaiveDate::parse_from_str(expires, "%Y%m%d").unwrap(), //.unwrap_or(Utc::now().naive_utc().add(Days::new(10))),
```

Comment:
Did you want the commented `unwrap_or` to be the behavior? 
Looks like the function `naive_utc()` gives a date+time rather than what the None branch gives. Using the same function as the other line gets rid of the type error: `chrono::NaiveDate::parse_from_str(expires, "%Y%m%d").unwrap_or(Utc::now().date_naive().add(Days::new(10))),`